### PR TITLE
renames var_2C to trackAndDirection where applicable

### PR DESCRIPTION
**Pull request by LeftofZen**
Just a bulk rename from `Vehicle*::var_2C` to `trackAndDirection`, since that's what var_2C is.

### CODE REVIEW

File: src/OpenLoco/Vehicles/Vehicle.h [hunk 1]
```diff
@@ -422,15 +422,15 @@ namespace OpenLoco::Vehicles
...
-        Map::SmallZ tileBaseZ;       // 0x34
-        uint8_t trackType;           // 0x35 field same in all vehicles
-        RoutingHandle routingHandle; // 0x36 field same in all vehicles
+        EntityId head;                       // 0x26
+        uint32_t remainingDistance;          // 0x28
+        TrackAndDirection trackAndDirection; // 0x2C
+        uint16_t subPosition;                // 0x2E
+        int16_t tileX;                       // 0x30
+        int16_t tileY;                       // 0x32
+        uint8_t tileBaseZ;                   // 0x34
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/Vehicle.h:431]: you've reverted the naming of the type of tileBaseZ to `Map::SmallZ`

**Comment by duncanspumpkin** [comment]
We really need to come up with a better name for it. TrackAndDirection doesn't really show how its fully used.

File: src/OpenLoco/Vehicles/Vehicle.h (hunk 1 above)
Comment (LeftofZen) [reply src/OpenLoco/Vehicles/Vehicle.h:431]: fuckin...I rebased to make sure I didn't revert these, not sure how this ended up like this :(

File: src/OpenLoco/Vehicles/Vehicle.cpp [hunk 2]
```diff
@@ -16,16 +16,16 @@ namespace OpenLoco::Vehicles
...
-        uint8_t trackType;           // 0x35 field same in all vehicles
-        RoutingHandle routingHandle; // 0x36 field same in all vehicles
-        uint8_t var_38;              // 0x38
+        EntityId head;                       // 0x26
+        uint32_t remainingDistance;          // 0x28
+        TrackAndDirection trackAndDirection; // 0x2C
+        uint16_t subPosition;                // 0x2E
+        int16_t tileX;                       // 0x30
+        int16_t tileY;                       // 0x32
+        uint8_t tileBaseZ;                   // 0x34
```
Comment (LeftofZen) [inline src/OpenLoco/Vehicles/Vehicle.cpp:25]: @duncanspumpkin this one actually WASN'T already SmallZ, should it be?

File: src/OpenLoco/Vehicles/Vehicle.h [hunk 3]
```diff
@@ -609,15 +609,15 @@ namespace OpenLoco::Vehicles
...
```
Comment (LeftofZen) [inline src/OpenLoco/Vehicles/Vehicle.h:618]: missed one

File: src/OpenLoco/Vehicles/Vehicle.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Vehicles/Vehicle.cpp:25]: Yes anything i wrote as tileBaseZ it was just to remind me that it is in SmallZ.

**Comment by LeftofZen** [comment]
> We really need to come up with a better name for it. TrackAndDirection doesn't really show how its fully used.

It's probably fine for now, but I'd imagine a simple change would be to split the 'track' part and the 'direction' part into their own uint8_t rather than merging the same uint16_t? Then again they are used together frequently. Otherwise, I'm not sure what I'd rename it to, any suggestions?

File: src/OpenLoco/Vehicles/Vehicle.cpp (hunk 2 above)
Comment (LeftofZen) [reply src/OpenLoco/Vehicles/Vehicle.cpp:25]: want me to change it in this PR or leave it to you?

Comment (duncanspumpkin) [reply src/OpenLoco/Vehicles/Vehicle.cpp:25]: Sure do it this pr

Comment (LeftofZen) [reply src/OpenLoco/Vehicles/Vehicle.cpp]: Done

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**